_posts/2025-02-03-Reduce-Cost-with-Disk-based-Vector-Search.md

```diff
@@ -0,0 +1,264 @@
+| `on_disk_16x` | 6               | r6gd.2xlarge   | 474                 | Instance            | 32 | 9    | 1                   | 16x            |
+| `on_disk_32x` | 4               | r6gd.2xlarge   | 474                 | Instance            | 32 | 6    | 1                   | 32x            |
+
+In the table, note that clusters with higher compression levels use significantly less resources than those without compression.
+
+To optimize performance for our tests, we made these additional adjustments:
+
+- Retrieved IDs from `doc_values` to reduce fetch time
+- Disabled `_source` storage 
+
+We ran all tests using OpenSearch Benchmark [vector search workloads](https://github.com/opensearch-project/opensearch-benchmark-workloads/tree/main/vectorsearch), following this procedure:
+
+1. Ingest the dataset.
+2. Force merge the index to 5 segments per shard.
+3. Run warm-up queries to load the index into memory.
+4. Run the single-client latency test.
+5. Run the multi-client throughput tests.
+6. Repeat steps 4 and 5 with disk-based rescore disabled to measure performance on a compressed index without rescoring.
+
+We obtained the following large-scale test results.
+
+| Metric/Configuration              | `in-memory` | `on_disk_8x` | `in_memory_8x` | `on_disk_16x` | `in_memory_16x` | `on_disk_32x` | `in_memory_32x` |
+|-----------------------------------|-----------|------------|--------------|-------------|---------------|-------------|---------------|
+| recall@100 (ratio)                | 0.95      | 0.98       | 0.98         | 0.97        | 0.96          | 0.94        | 0.95          |
+| 1-client p90 search latency (ms)  | 24.02     | 96.31      | 28.90        | 108.05      | 29.79         | 104.421     | 47.2906       |
+| 1-client mean throughput (QPS)           | 40.64     | 11.03      | 42.33        | 10.06       | 43.95         | 10.65       | 31.83         |
+| 4-client p90 search latency (ms)  | 25.82     | 97.19      | 20.40        | 220.19      | 18.09         | 244.52      | 46.13         |
+| 4-client mean throughput (QPS)           | 162.80    | 45.86      | 204.62       | 25.80       | 193.27        | 25.28       | 146.77        |
+| 8-client p90 search latency (ms) | 27.69     | 95.05      | 30.88        | 414.20      | 27.94         | 429.79      | 25.07         |
+| 8-client mean throughput (QPS)           | 306.70    | 95.22      | 305.86       | 26.34       | 343.95        | 25.75       | 376.60        |
+
+Interestingly, for this dataset, the on-disk approach with rescoring produces similar recall to the in-memory approach without rescoring, but the in-memory approach is substantially faster. This is most likely because the Cohere v3 model has been optimized to work very well with binary quantized data (see [this blog post](https://cohere.com/blog/int8-binary-embeddings)).
+
+## Learnings
+
```

Comment:
Line 260: We haven't actually referenced ANN prior to this. Instead of "ANN approach", do we mean "nearest neighbor approach"?